deps/Makefile

```diff
@@ -217,22 +223,53 @@ $(foreach dir,$(DIRS),$(eval $(call dir_target,$(dir))))
 $(build_prefix): | $(DIRS)
 $(eval $(call dir_target,$(SRCDIR)/srccache))
 
+upper = $(shell echo $1 | tr a-z A-Z)
+
 ## A rule for calling `make install` ##
-#	rule: dependencies
-#   	$(call make-install,rel-build-directory,add-args)
-#   	touch -c $@
+# example usage:
+#   $(call staged-install, \
+#       1 target, \               # name
+#       2 rel-build-directory, \  # BUILDDIR-relative path to binaries
+#       3 MAKE_INSTALL, \         # will be called with args SRCDIR,DESTDIR,$4
+#       4 add-args, \             # extra args for $3
+#       5 target-file, \          # installation-target file
+#       6 post-install)           # post-install commands
+#
 # this rule ensures that make install is more nearly atomic
 # so it's harder to get half-installed (or half-reinstalled) dependencies
-MAKE_DESTDIR = "$(build_staging)/$1"
-define staged-install
-	rm -rf $(build_staging)/$1
-	$2
-	mkdir -p $(build_prefix)
-	cp -af $(build_staging)/$1$(build_prefix)/* $(build_prefix)
+# and enables sharing deps compiles, uninstall, and fast reinstall
+MAKE_INSTALL = +$$(MAKE) -C $1 install $$(MAKE_COMMON) $3 DESTDIR="$2"
+define LIBFILE_INSTALL
+	mkdir -p $2/$$(build_libdir)
+	cp $3 $2/$$(build_libdir)
```

Comment:
i can't give then nice variable names, but `$2` is a newly created, pristine DESTDIR